leshan-core/src/main/java/org/eclipse/leshan/core/node/codec/senml/LwM2mNodeSenMLJsonEncoder.java

```diff
@@ -0,0 +1,191 @@
+            for (LwM2mResource resource : instance.getResources().values()) {
+                // Validate request path & compute resource path
+                String prefixPath;
+                if (requestPath.isObject()) {
+                    prefixPath = instance.getId() + "/" + resource.getId();
+                } else if (requestPath.isObjectInstance()) {
+                    prefixPath = Integer.toString(resource.getId());
+                } else {
+                    throw new CodecException("Invalid request path %s for instance encoding", requestPath);
+                }
+                // Create resources
+                lwM2mResourceToSenMLRecord(prefixPath, timestamp, resource);
+            }
+        }
+
+        @Override
+        public void visit(LwM2mResource resource) {
+            LOG.trace("Encoding resource {} into SenML JSON", resource);
+            if (!requestPath.isResource()) {
+                throw new CodecException("Invalid request path %s for resource encoding", requestPath);
+            }
+
+            lwM2mResourceToSenMLRecord(EMPTY_RESOURCE_PATH, timestamp, resource);
+        }
+
+        private void lwM2mResourceToSenMLRecord(String resourcePath, Long timestamp, LwM2mResource resource) {
+            // create resource element
+            if (resource.isMultiInstances()) {
+                for (Entry<Integer, ?> entry : resource.getValues().entrySet()) {
+                    // compute resource instance path
+                    String resourceInstancePath;
+                    if (resourcePath == null || resourcePath.isEmpty()) {
+                        resourceInstancePath = Integer.toString(entry.getKey());
+                    } else {
+                        resourceInstancePath = resourcePath + "/" + entry.getKey();
+                    }
+
+                    addSenMLRecord(resourceInstancePath, timestamp, resource, entry.getValue());
+                }
+            } else {
+                addSenMLRecord(resourcePath, timestamp, resource, resource.getValue());
+            }
+        }
+
+        private void addSenMLRecord(String resourcePath, Long timestamp, LwM2mResource resource, Object value) {
+            // get type for this resource
+            ResourceModel rSpec = model.getResourceModel(objectId, resource.getId());
+            Type expectedType = rSpec != null ? rSpec.type : resource.getType();
+
+            // Create resource element
+            SenMLRecord record = new SenMLRecord();
+            
+            if (records.isEmpty() && requestPath.isObjectInstance()) {
+                record.setBaseName(requestPath.toString());
+            }
+            
+            if(requestPath.isResource()) {
+                record.setBaseName(requestPath.toString());
+            }
+            else if(requestPath.isObjectInstance()) {
+                record.setName(resourcePath);
+                record.setTime(timestamp);
+            }
+
```

Comment:
I suspect the way, setBaseName and Name is not correct ? What happens for `Object` path ?

A first minimal way could looks like this ?
```java
 // We set basename only on first record.
if (records.isEmpty()) {
    record.setBaseName(requestPath.toString() + "/");
}

// Set name if present
if (resourcePath != null && !resourcePath.isEmpty()) {
    record.setName(resourcePath);
}
```